src/client/src/apps/SimpleLauncher/Launcher.tsx

```diff
@@ -1,119 +1,135 @@
-          <LogoContainer>
-            <LogoIcon width={1.3} height={1.3} />
-          </LogoContainer>
-          {appConfigs.map(app => (
-            <ButtonContainer key={app.name}>
-              <LaunchButton onClick={() => open(app)}>
-                {app.icon}
-                <IconTitle>{app.name}</IconTitle>
-              </LaunchButton>
-            </ButtonContainer>
-          ))}
-          <LauncherExit />
-          <ThemeSwitchContainer>
-            <ThemeStorageSwitch />
-          </ThemeSwitchContainer>
-        </Root>
-      </React.Fragment>
-    )
-  }
-}
 
-const Root = styled.div`
-  height: 100%;
-  width: 100%;
-  display: flex;
-  align-items: center;
-  justify-content: center;
-  background-color: ${({ theme }) => theme.core.lightBackground};
-  color: ${({ theme }) => theme.core.textColor};
-`
+const SearchButton: React.FC<{ onClick: () => void }> = ({ onClick }) => (
+  <ButtonContainer>
+    <LaunchButton onClick={onClick}>{SearchIcon}</LaunchButton>
+  </ButtonContainer>
+)
+
+export const Launcher: React.FC = () => {
+  const [initialBounds, setInitialBounds] = useState<Bounds>()
+  const [isSearchVisible, setIsSearchVisible] = useState<boolean>(false)
+  const [isSearchBusy, setIsSearchBusy] = useState<boolean>()
+  const searchInputRef = useRef<HTMLInputElement>(null)
+
+  useEffect(() => {
+    getCurrentWindowBounds().then(setInitialBounds)
+  }, [])
+
+  const showSearch = useCallback(
+    () => {
+      if (isSearchVisible) {
+        searchInputRef.current && searchInputRef.current.focus({ preventScroll: true })
+      }
+      setIsSearchVisible(true)
+    },
+    [isSearchVisible]
+  );
+
+  // if search is made visible - focus on it
+  useEffect(
+    () => {
+      if (isSearchVisible) {
+        searchInputRef.current && searchInputRef.current.focus({ preventScroll: true })
+      }
+    },
+    [isSearchVisible]
+  )
 
-const IconTitle = styled.span`
-  position: absolute;
-  bottom: 2px;
-  right: 0;
-  left: 0;
-  text-align: center;
-  font-size: 9px;
-  font-family: Lato;
-  color: transparent;
-  transition: color 0.3s ease;
+  // hide search if Escape is pressed
+  useEffect(
+    () => {
+      const onKeyDown = (event: KeyboardEvent) => {
+        if (event.key === 'Escape') {
+          setIsSearchVisible(false)
+        }
+      }
+      document.addEventListener('keydown', onKeyDown)
+      return () => document.removeEventListener('keydown', onKeyDown)
+    }, []
+  )
 
-  /* avoids text highlighting on icon titles */
-  user-select: none;
-`
+  const handleSearchStateChange = useCallback(
+    (state: SearchState) => setIsSearchBusy(state.typing || state.loading),
+    []
+  )
 
-const IconContainer = styled.div`
-  height: 100%;
-  width: 100%;
-  display: flex;
-  align-items: center;
-  justify-content: center;
-  flex-direction: column;
-  &:hover {
-    span {
-      color: ${({ theme }) => theme.core.textColor};
-    }
-  }
-`
+  const handleSearchSizeChange = useCallback(
```

Comment:
this is where we know that search component changed in size and thus we need to change window size